src/App.js

```diff
@@ -69,14 +71,16 @@ function App() {
     let timer;
     if (isGameStarted) {
       timer = setInterval(() => {
-        setSquares(prevSquares => {
-          const randomIndex = Math.floor(Math.random() * prevSquares.length);
+        setSquares((prevSquares) => {
+          const unToggled = prevSquares.filter((obj) => !obj.on);
```

Comment:
@deonneon so this makes it so it will only select from the boxes that are untoggled. this is the root of the bug where it never finishes since at the end the probability of it correctly getting the single untoggled box was low